test/sql/test_substrait_nested_types.test

```diff
@@ -0,0 +1,46 @@
+# name: test/sql/test_substrait_nested_types.test
+# description: Test nested type support in substrait extension (LIST, MAP, ARRAY)
```

Comment:
What's the difference between a list and an array?  Is STRUCT supposed to be listed here?

Reply:
Ah sorry, leftover... I tried to implement the ARRAY type (fixed-length list basically), and reverted it because I didn't have a clear use case for it, and it was harder to test due to Substrait not having fixed-length arrays.

I'll remove ARRAY from the comment here.